llumnix/backends/vllm/llm_engine.py

```diff
@@ -157,9 +146,13 @@ def update_instance_info(self, instance_info: InstanceInfo) -> None:
             instance_info.num_blocks_last_running_request = self.instance_info.num_blocks_last_running_request
         self.instance_info = instance_info
 
-    def _put_request_output_to_server(self,
-                                      request_outputs: List[RequestOutput],
-                                      server_infos: List[ServerInfo]) -> None:
+    def add_request(self, request_id: str, server_info: ServerInfo, *args, **kwargs):
+        super().add_request(request_id, *args, **kwargs)
+        seq_group = self.scheduler.waiting[-1]
+        self.scheduler.waiting[-1] = SequenceGroupLlumnix(request_id, server_info, [seq_group.get_seqs()[0]], seq_group.sampling_params,
+                                  seq_group.metrics.arrival_time, seq_group.lora_request, seq_group.multi_modal_data)
```

Comment:
If seq_group is a member variable, We need rewrite all functions in SequenceGroup. Actually use monkey patching like`vllm.sequence.SequenceGroup = SequenceGroupLlumnix` is a better solution, but I it doesn't work because `vllm.sequence` has static class and use monkey patching will cause bugs.